i18n-helpers/src/xgettext.rs

```diff
@@ -257,79 +233,184 @@ struct Chapter {
+    }
+
+    /// Mimics the `push` and `pop` functionalities of `std::path::PathBuf`
+    /// with, respectively, the [`maybe_push`] and [`maybe_pop`] methods,
+    /// so that:
+    ///
+    /// 1. The path is never extended past `max_depth` depth.
+    /// 2. Whenever [`maybe_push`] is invoked, the base directory/file of
+    ///    the path is potentially altered to make the path unique among
+    ///    the ones produced thus far.
+    /// 3. All strings passed to [`maybe_push`] are normalized by retaining
+    ///    only non-alphabetic characters and hyphens.
+    ///
+    /// The disambiguation is deterministic (i.e., the same sequence of
+    /// operation is guaranteed to always produce the same paths) and not
+    /// based on randomization, therefore its stability doesn't depend on
+    /// the stability of external libraries.
+    ///
+    /// The [`get`] method returns the current path with a .pot extension.
+    ///
+    /// The spacial case of `depth` equal to 0 is treated differently by
+    /// returning the `DEFAULT_DEPTH_0`.pot filename.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use mdbook_i18n_helpers::xgettext::paths::UniquePathBuilder;
+    ///
+    ///
+    /// let mut unique_path_builder = UniquePathBuilder::new(2);
+    ///
+    /// fn get_str(pb: &UniquePathBuilder) -> String {
+    ///     pb.get().to_str().unwrap().into()
+    /// }
+    ///
+    /// unique_path_builder.maybe_push("foo"); // Pushes "foo"
+    /// unique_path_builder.maybe_push("bar"); // Pushes "bar"
+    /// assert_eq!(get_str(&unique_path_builder), "foo/bar.pot");
+    ///
+    /// unique_path_builder.maybe_push("baz"); // No-op for the user
+    /// assert_eq!(get_str(&unique_path_builder), "foo/bar.pot");
+    ///
+    /// unique_path_builder.maybe_pop(); // No-op
+    /// unique_path_builder.maybe_pop(); // Pops "bar"
+    /// unique_path_builder.maybe_pop(); // Pops "foo"
+    /// unique_path_builder.maybe_push("foo"); // Pushes "foo_1"
+    /// unique_path_builder.maybe_push("bar"); // Pushes "bar"
+    /// unique_path_builder.maybe_pop(); // Pops "bar"
+    /// unique_path_builder.maybe_push("bar"); // Pushes "bar_1"
+    /// assert_eq!(get_str(&unique_path_builder), "foo_1/bar_1.pot");
+    /// ```
+    impl UniquePathBuilder {
+        const DEFAULT_DEPTH_0: &str = "messages";
+
+        pub fn new(max_depth: usize) -> Self {
+            Self {
+                current_path: PathBuf::new(),
+                current_depth: 0,
+                max_depth,
+                frequency_map: HashMap::new(),
+            }
+        }
+
+        fn format_path_with_counter(&self) -> PathBuf {
+            match self.frequency_map.get(&self.current_path) {
+                None | Some(&0) => self.current_path.clone(),
+                Some(&cnt) => {
+                    let mut file_name = self.current_path.file_name().unwrap().to_os_string();
+                    file_name.push(format!("_{}", cnt));
```

Comment:
I prefer `-` over `_` in filenames, where possible. So could we do it like this instead:

```suggestion
                    file_name.push(format!("-{cnt}"));
```

Reply:
Sure, changed!